graphql/util.go

```diff
@@ -543,6 +544,35 @@ func generateBuildVariants(sc data.Connector, versionId string, searchVariants [
 	return result, nil
 }
 
+// getFailedTestResultsSample returns a sample of failed test results for the given tasks that match the supplied testFilters
+func getCedarFailedTestResultsSample(ctx context.Context, tasks []task.Task, testFilters []string) ([]apimodels.CedarFailedTestResultsSample, error) {
+	if len(tasks) == 0 {
+		return nil, nil
+	}
+	cedarTaskInfo := []apimodels.CedarTaskInfo{}
+	for _, t := range tasks {
+		cedarTestOption := apimodels.CedarTaskInfo{
+			TaskID:      t.Id,
+			Execution:   t.Execution,
+			DisplayTask: t.DisplayOnly,
+		}
+		cedarTaskInfo = append(cedarTaskInfo, cedarTestOption)
+	}
+	cedarFailedTestSampleOpts := apimodels.CedarFailedTestSampleOptions{
+		Tasks:        cedarTaskInfo,
+		RegexFilters: testFilters,
+	}
+	opts := apimodels.GetCedarFailedTestResultsSampleOptions{
```

Comment:
Nit: you can combine this with the three lines above instead of making a separate variable.
```
opts := apimodels.GetCedarFailedTestResultsSampleOptions{
        BaseURL: evergreen.GetEnvironment().Settings().Cedar.BaseURL,
        SampleOptions: apimodels.CedarFailedTestSampleOptions{
                Tasks: cedarTaskInfo,
                RegexFilters: testFilters,
        },
}
```